learning-another-language/js-loops.md

```diff
@@ -0,0 +1,417 @@
+  </summary>
+
+  ```javascript
+    let i = 0;
+    while (i <= 20) {
+        console.log(i);
+        i += 5;
+    }
+
+    // Output:
+    // 0
+    // 5
+    // 10
+    // 15
+    // 20
+  ```
+</details>
+
+## 5. do…while loop
+```js
+do {
+	// perform repetitive task at least once
+} while(condition)
+```
+
+The `do...while` statement creates a loop that executes a block of code until a condition evaluates to `false`.
+
+The statements in the block will **always** be executed once before the condition is checked. This is what differentiates a `do...while` loop from a `while` loop. The `do...while` loop is sometimes called a post-test loop.
+
+### Example
+```js
+let i = 1
+do {
+	console.log(i);
+	i += 1;
+} while (i <= 5)
+
+// Output:
+// 1
+// 2
+// 3
+// 4
+// 5
+```
+
+### Common Pitfall: Indefinite Loop
+
+It’s important to ensure the variable(s) used in the statement are updated such that the `while(condition)` evaluates to false. Otherwise, you may crash your program due to an infinite loop.
+
+Spot the bug below:
+```js
+let x = 5;
+let y = 4;
+let sum = 0
+do {
+	sum = x + y;
+	console.log(sum);
+	x = x * x
+	y = y * y
+} while(sum >= 0);
+```
+
+<details style="max-width: 700px; margin: auto;">
+  <summary>Click here for an explanation</summary>
+
+  The program will run indefinitely because the condition `sum >= 0` will always evaluate to `true`. We can fix this by changing the condition to `sum <= 100` or anything that will eventually evaluate to `false`.
+</details>
+
+## 6. Additional Statements
+### forEach
+```js
+myArray.forEach(myFunction(currentValue, index, arr))
+```
+
+The `forEach` method calls a function and iterates over the elements of an array, map, or set. 
+
+The syntax  for the `forEach` method has multiple parts, some of them being optional:
+
+- `myArray` is the name of the array through which items you would like to iterate.
+- `myFunction` is the name of the function you would like to run for each of the elements in the array.
+- `currentValue` an element in the array which is currently being iterated upon. It is the only variable which is required to be passed into the function `myFunction`. This is the variable that will represent each element in the array.
+- `index` is an **optional** variable used to indicate the index of the current element in the array which is being iterated upon.
+- `arr` is an **optional** variable which is the array of the current elements.
+
+### Example
+```js
+let instructors = ["Auberon", "Char", "Claire", "Kyra"];
+
+function sayHello(instructor) {
+	console.log(`Hello, ${instructor}!`);
+}
+
+instructors.forEach(sayHello);
+
+// Output:
+// Hello, Auberon!
+// Hello, Char!
+// Hello, Claire!
+// Hello, Kyra!
+
+function addHello(instructor, index, arr) {
+	arr[index] = 'Hello, ' + instructor;
+}
+
+instructors.forEach(addHello);
+
+console.log(instructors)
+
+// Output:
+// ['Hello, Auberon', 'Hello, Char', 'Hello, Claire', 'Hello, Kyra']
+```
+
+### break
```

Comment:
Consider adding a paragraph on coding style and `break` and `continue` statements. While they can be useful, we often want to minimize their usage because they can make the execution flow unpredictable/difficult to follow.